devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy

```diff
@@ -133,20 +179,30 @@ private handleBatteryEvent(volts) {
 }
 
 private handlePresenceEvent(present) {
-    def wasPresent = device.currentState("presence")?.value == "present"
-    if (!wasPresent && present) {
-        log.debug "Sensor is present"
-        startTimer()
-    } else if (!present) {
-        log.debug "Sensor is not present"
-        stopTimer()
+    if (isVision()) {
+        if (!state.gsensor && present) {
+            log.debug "Vision Sensor is present"
+            startTimer()
+        } else if (!present) {
+            log.debug "Vision Sensor is not present"
+            stopTimer()
+        }
+    } else {
+        def wasPresent = device.currentState("presence")?.value == "present"
+        if (!wasPresent && present) {
+            log.debug "Sensor is present"
+            startTimer()
+        } else if (!present) {
+            log.debug "Sensor is not present"
+            stopTimer()
+        }
     }
     def linkText = getLinkText(device)
     def descriptionText
     if ( present )
```

Comment:
Please, could You add parentheses to this block ?

Reply:
OK! I will add parentheses to this block.